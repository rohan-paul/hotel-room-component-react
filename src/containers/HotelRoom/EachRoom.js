import React from "react"
import { useSelector, useDispatch } from "react-redux"
import { Typography } from "@material-ui/core"
import IconButton from "@material-ui/core/IconButton"
import { useStyles } from "./EachRoomStyles"

const EachRoom = ({
  rooms,
  adults,
  children,
  leftText,
  incrementRoomsNo,
  decrementRoomsNo,
  incrementAdultsNo,
  decrementAdultsNo,
  incrementChildrenNo,
  decrementChildrenNo,
}) => {
  const globalStore = useSelector(state => state.globalStore)
  const classes = useStyles()
  const dispatch = useDispatch()
  const { no_of_rooms, no_of_adults, no_of_children } = globalStore

  const handleIncrementClick = () => {
    if (rooms) {
      dispatch(incrementRoomsNo())
    } else if (adults) {
      dispatch(incrementAdultsNo())
    } else if (children) {
      dispatch(incrementChildrenNo())
    }
  }

  const handleDecrementClick = () => {
    if (rooms) {
      dispatch(decrementRoomsNo())
    } else if (adults) {
      dispatch(decrementAdultsNo())
    } else if (children) {
      dispatch(decrementChildrenNo())
    }
  }

  const maxPersons = no_of_rooms * 4

  return (
    <>
      <div className={classes.tableContainer}>
        <div className={classes.roomsContainer}>
          <IconButton
            aria-label="Filter list"
            classes={{ root: classes.button, disabled: classes.buttonDisabled }}
          >
            <img
              src={
                adults
                  ? require("../../assets/images/adult-icon.png")
                  : children
                  ? require("../../assets/images/child-icon.png")
                  : rooms
                  ? require("../../assets/images/bed-icon.png")
                  : null
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
          <div className={classes.leftTextContainer}>
            <Typography variant="subtitle1">{leftText}</Typography>
          </div>
        </div>
        <div className={classes.ctaContainer}>
          <IconButton
            aria-label="Filter list"
            className={classes.adultIconContainer}
            disabled={
              (rooms && no_of_rooms <= 1) ||
              (adults && no_of_adults <= 1) ||
              (children && no_of_children <= 0)
            }
            onClick={handleDecrementClick}
          >
            <img
              src={
                (rooms && no_of_rooms <= 1) ||
                (adults && no_of_adults <= 1) ||
                (children && no_of_children <= 0)
                  ? require("../../assets/images/minus-icon-disabled.png")
                  : require("../../assets/images/minus-icon.png")
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
          <div className={classes.leftTextContainer}>
            <Typography variant="subtitle1">
              {rooms
                ? no_of_rooms
                : adults
                ? no_of_adults
                : children
                ? no_of_children
                : null}
            </Typography>
          </div>
          <IconButton
            aria-label="Filter list"
            className={classes.adultIconContainer}
            onClick={handleIncrementClick}
            disabled={
              (rooms && no_of_rooms >= 5) ||
              (adults && no_of_adults + no_of_children >= maxPersons) ||
              (children && no_of_adults + no_of_children >= maxPersons)
            }
          >
            <img
              src={
                (rooms && no_of_rooms >= 5) ||
                (adults && no_of_adults + no_of_children >= maxPersons) ||
                (children && no_of_adults + no_of_children >= maxPersons)
                  ? require("../../assets/images/plus-icon-disabled.png")
                  : require("../../assets/images/plus-icon.png")
              }
              className={classes.avatar}
              alt=""
            />
          </IconButton>
        </div>
      </div>
    </>
  )
}

export default EachRoom
