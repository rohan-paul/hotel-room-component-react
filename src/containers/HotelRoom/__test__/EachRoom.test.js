import React from "react"
import renderer from "react-test-renderer"
import { create } from "react-test-renderer"
import { MuiThemeProvider } from "@material-ui/core/styles"
import globalTheme from "../../../globalTheme"
import PropTypes from "prop-types"
import { mount, shallow } from "enzyme"
import expect from "expect"
import { createShallow, createMount } from "@material-ui/core/test-utils"
import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import { Provider } from "react-redux"
import EachRoom from "../EachRoom"
import { render, fireEvent } from "@testing-library/react"

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const store = mockStore({
  globalStore: {
    no_of_rooms: 1,
    no_of_adults: 2,
    no_of_children: 0,
  },
})

describe("EachRoom Component", () => {
  it("should match snapshot", () => {
    const matches = (children, theme = globalTheme) =>
      expect(
        renderer
          .create(
            <MuiThemeProvider theme={globalTheme}>{children}</MuiThemeProvider>,
          )
          .toJSON(),
      ).toMatchSnapshot()
  })

  it("should should match snapshot", () => {
    const wrapper = create(
      <Provider store={store}>
        <MuiThemeProvider theme={globalTheme}>
          <EachRoom />
        </MuiThemeProvider>
      </Provider>,
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
