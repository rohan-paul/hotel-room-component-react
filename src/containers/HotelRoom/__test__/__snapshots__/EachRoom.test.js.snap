// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EachRoom Component should render a startup component if startup is not complete 1`] = `
<div
  className="makeStyles-tableContainer-2"
>
  <div
    className="makeStyles-roomsContainer-5"
  >
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Click Me"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src={null}
        />
      </span>
    </button>
    <div
      className="makeStyles-leftTextContainer-7"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      />
    </div>
  </div>
  <div
    className="makeStyles-ctaContainer-6"
  >
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src="minus-icon.png"
        />
      </span>
    </button>
    <div
      className="makeStyles-leftTextContainer-7"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      />
    </div>
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src="plus-icon.png"
        />
      </span>
    </button>
  </div>
</div>
`;

exports[`EachRoom Component should should match snapshot 1`] = `
<div
  className="makeStyles-tableContainer-2"
>
  <div
    className="makeStyles-roomsContainer-5"
  >
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Click Me"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src={null}
        />
      </span>
    </button>
    <div
      className="makeStyles-leftTextContainer-7"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      />
    </div>
  </div>
  <div
    className="makeStyles-ctaContainer-6"
  >
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src="minus-icon.png"
        />
      </span>
    </button>
    <div
      className="makeStyles-leftTextContainer-7"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      />
    </div>
    <button
      aria-label="Filter list"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <img
          alt=""
          className="makeStyles-avatar-4"
          src="plus-icon.png"
        />
      </span>
    </button>
  </div>
</div>
`;
